kubelet/datadog_checks/kubelet/kubelet.py

```diff
@@ -636,9 +636,8 @@ def _update_kubelet_url_and_bearer_token(self, instance, endpoint):
         self.update_prometheus_url(instance, self.kubelet_scraper_config, kubelet_metrics_endpoint)
 
         probes_metrics_endpoint = urljoin(endpoint, PROBES_METRICS_PATH)
-        if self.detect_probes(self.get_http_handler(self.probes_scraper_config), probes_metrics_endpoint):
```

Comment:
question: is the issue that the head request in the `detect_probes` function is not returning 200? would that cause an issue as well here: https://github.com/DataDog/integrations-core/blob/15befe2aecd6c7608639c329b986b43eef7f2915/kubelet/datadog_checks/kubelet/kubelet.py#L359-L361

Reply:
No, the issue is that we are trying to make the request before the prometheus url can use the updated value fetched from `get_connection_info()`, and before `self.kubelet_credentials.configure_scraper(self.probes_scraper_config)` is called.

The line that you highlighted happens after these 2 steps have happened